File: 0633_sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        if (c == 0) return true;
        for (long a = 1; a <= c; a++) {
            long aa = a * a;
            long bb = c - aa;
            if (bb < 0) {
                return false;
            }
            long b = sqrt(bb);
            if (aa + b * b == c) return true;
        }
        return false;
    }
};

File: 0633_sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        lo, hi = 0, int(c**0.5)
        while lo <= hi:
            cc = lo*lo + hi*hi
            if cc == c:
                return True
            elif cc > c:
                hi -= 1
            else:
                lo += 1
        return False
